src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +279,837 @@ static DamageTable D_808792D4 = {
+        sp30.y = this->collider1Elements[2].dim.worldSphere.center.y;
+        sp30.z = this->collider1Elements[2].dim.worldSphere.center.z;
+        func_80876930(this, globalCtx, &sp30);
+    } else if (func_801378B8(&this->skelAnime, 39.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_J_WALK);
+        sp30.x = this->collider1Elements[1].dim.worldSphere.center.x;
+        sp30.y = this->collider1Elements[1].dim.worldSphere.center.y;
+        sp30.z = this->collider1Elements[1].dim.worldSphere.center.z;
+        func_80876930(this, globalCtx, &sp30);
+    }
+
+    if ((Math_Vec3f_DistXZ(&this->actor.home.pos, &player->actor.world.pos) < 400.0f) &&
+        (Player_GetMask(globalCtx) != PLAYER_MASK_STONE_MASK)) {
+        temp_v1 = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+        Math_ScaledStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 0x1F4);
+        if (this->actor.xzDistToPlayer < (100.0f * this->unk_334)) {
+            if ((ABS_ALT(temp_v1) < 0x1388) && (this->actor.yDistToPlayer < 60.0f) &&
+                !(player->stateFlags1 & 0x800000)) {
+                func_808777A8(this);
+            }
+        }
+    } else {
+        if ((Actor_XZDistanceToPoint(&this->actor, &this->actor.home.pos) > 150.0f) ||
+            (Player_GetMask(globalCtx) == PLAYER_MASK_STONE_MASK)) {
+            Math_ScaledStepToS(&this->actor.world.rot.y, Actor_YawToPoint(&this->actor, &this->actor.home.pos), 0x1F4);
+        }
+
+        this->timer--;
+        if (this->timer == 0) {
+            if (Rand_ZeroOne() > 0.7f) {
+                this->timer = Rand_S16Offset(50, 70);
+            } else {
+                func_808773C4(this);
+            }
+        }
+    }
+
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+}
+
+void func_808777A8(EnDodongo* this) {
+    s32 i;
+    Sphere16* sph;
+
+    SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_060028F0, -4.0f);
+    this->actor.speedXZ = 0.0f;
+
+    for (i = 0; i < ARRAY_COUNT(this->collider3Elements); i++) {
+        sph = &this->collider3.elements[i].dim.worldSphere;
+
+        sph->center.x = this->unk_348[0].x;
+        sph->center.y = this->unk_348[0].y;
+        sph->center.z = this->unk_348[0].z;
+        sph->radius = 0;
+    }
+
+    this->actionFunc = func_8087784C;
+}
+
+void func_8087784C(EnDodongo* this, GlobalContext* globalCtx) {
+    static Vec3f D_80879348 = { 0.0f, 0.9f, 0.0f };
+    static Vec3f D_80879354 = { 0.0f, 0.0f, 0.0f };
+    s16 frame;
+    ColliderJntSphElement* element;
+    s32 i;
+    s32 end;
+    f32 temp_f2;
+    f32 temp_f12;
+
+    if (func_801378B8(&this->skelAnime, 24.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_J_CRY);
+    }
+
+    if (func_8087721C(this)) {
+        func_800B9010(&this->actor, NA_SE_EN_DODO_J_FIRE - SFX_FLAG);
+        frame = this->skelAnime.animCurrentFrame - 29.0f;
+        end = frame >> 1;
+        if (end >= 4) {
+            end = 3;
+        }
```

Comment:
```suggestion
        if (end > 3) {
            end = 3;
        }
```
Could this also be a CLAMP_MAX?